q.cpp

```diff
@@ -0,0 +1,245 @@
+#include <stdio.h>
+#include <string.h>
+#include <malloc.h>
```

Comment:
By the way, how do you compile it? `makefile` has not been modified in this PR, so I guess that `make q` wouldn't work...